yesqa.py

```diff
@@ -49,12 +49,26 @@ def _remove_comments(tokens: Tokens) -> Tokens:
     for i, token in tokenize_rt.reversed_enumerate(tokens):
         if token.name == 'COMMENT':
             if NOQA_RE.search(token.src):
-                _rewrite_noqa_comment(tokens, i, collections.defaultdict(set))
+                _mask_noqa_comment(tokens, i)
             elif NOQA_FILE_RE.search(token.src):
                 _remove_comment(tokens, i)
     return tokens
 
 
+def _mask_noqa_comment(tokens: Tokens, i: int) -> None:
+    token = tokens[i]
+    match = NOQA_RE.search(token.src)
+    assert match is not None
+    if match.group() == token.src:
+        tokens[i] = tokens[i]._replace(src=f'# {"."*(len(token.src)-2)}')
+    else:
+        src = NOQA_RE.sub(
+            lambda x: f'# {"."*(len(x.group())-2)}',
```

Comment:
I would write this as a separate function so the `x` can be typed -- and there's no need for the two branches here since the second one handles the cases as well